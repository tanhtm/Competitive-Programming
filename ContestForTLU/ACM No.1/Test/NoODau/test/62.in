2953 66
4 231 319 1039 1618 1691 1857 2150 2613 2674 2707 2950 3166 5048 5344 6069 6112 6294 6307 6434 6807 7581 7857 8787 9714 10117 10175 10191 10257 11413 11462 12118 12204 12271 12487 13153 13457 14159 14169 14390 14586 14692 15151 15398 15467 15731 15754 15776 16032 18004 18103 18192 18333 18443 18974 19342 19440 19589 19864 19952 20159 20255 21325 22035 22717 23074 23276 23347 23368 23767 24332 24338 24509 24630 25197 25367 25440 26472 26528 27741 29230 29255 29403 29937 30072 30259 30601 30732 31184 31440 31700 31903 32189 32319 32544 32998 33099 33481 33500 33616 33977 34781 34807 35212 35384 36074 36521 36692 37246 37388 38044 38795 38837 38896 39208 39315 39534 39603 39618 40479 40902 41440 42005 42045 42087 42165 42473 43328 43534 43539 43708 43731 43824 44157 44224 44352 44801 44885 44933 45663 45702 45902 46080 46539 46981 49094 49135 49409 49565 49825 49835 49982 50283 50483 50623 51224 52212 52301 52303 52685 53463 53932 54283 55374 56357 56391 57954 58017 58389 58421 59048 59129 59649 59686 59736 60129 61621 61633 61702 61744 61778 62590 62657 62859 62863 62907 63017 63069 63744 65030 65742 66025 66443 66656 66683 67085 67163 67265 67417 67715 68136 68796 69371 69584 69652 69708 69732 70343 70933 71685 72770 73278 73414 73671 73919 75741 75954 76740 77172 77746 77747 78320 78536 78792 78856 78891 79595 79733 80057 80372 80427 80686 81494 82675 82907 83621 83785 84100 84408 84454 84527 85047 85180 85329 86322 86389 87425 87505 87515 88222 88458 88632 88805 88815 88930 89119 89634 89882 91588 92039 92154 92679 92688 93156 93319 93333 94133 94209 94336 94674 94737 95698 95911 95959 96611 96636 97151 97158 98047 98255 98558 98628 99286 100259 100381 100393 100654 101287 101550 101786 101803 102495 102644 103303 103597 103722 104163 105095 105154 105436 105572 105761 105844 107772 108261 108790 109123 109181 109386 109388 109635 109698 110019 110530 110992 111007 111342 111355 111359 111386 111961 112390 112806 112878 112985 113885 114199 114317 115113 115303 115310 115657 115723 116074 116169 116313 116468 116533 116887 116933 117072 117352 117355 117406 118918 118988 119469 119934 120066 120204 120318 120449 120607 121036 121055 121921 122014 122243 122301 122718 122768 122831 122959 123460 123528 123543 123589 123717 123784 123928 124169 124220 124257 124487 124854 124901 125710 126059 126142 126661 127230 127640 127671 127841 128268 129147 129243 129418 129657 129674 130530 130614 130715 131552 131673 132359 132897 134021 134489 134814 135073 135082 135486 135502 135560 135754 136087 136907 138071 138312 138346 138825 138961 139602 139838 140898 140962 141372 141635 141955 142506 143129 143139 143778 144005 144537 144674 145689 145854 146030 147151 147432 148728 149056 149129 149172 149667 150326 150425 150432 150899 151394 151658 152202 152920 153457 153507 153841 153847 153980 154187 154504 155643 155873 156031 156253 156349 156453 157139 157455 157475 157802 158075 158085 158840 158879 159375 159391 159477 159526 159849 160233 161144 161691 161776 161806 162214 162510 162723 162868 163363 163459 163589 164190 164361 164604 165150 166073 166127 166144 166303 166308 167246 167525 167565 167601 167701 167824 168974 169004 169070 169385 169435 171478 172274 172803 172933 172979 173319 174079 174178 174273 174354 174514 174530 174713 175381 175618 175690 177036 177092 177676 177833 178491 178525 178892 179264 179269 179422 179561 179998 180026 180157 180453 180485 180599 180721 181393 181670 181884 181900 181970 182047 182130 182499 182964 183263 183454 184522 184578 185072 185349 185383 185438 185986 186025 186101 186233 186407 186438 186528 186830 187027 187853 188811 188947 189012 189190 189491 189503 189538 189643 190370 191378 191538 191664 192516 192778 192794 193170 193578 194070 195197 195202 195349 196209 196345 196463 196552 197326 197562 198034 198112 198171 198295 198536 198649 199026 199092 199323 199633 200008 200161 200256 200275 200513 200740 200751 201331 201364 201621 201658 201966 202245 202477 202498 202550 203804 204463 204720 205013 205335 205741 205944 206421 206935 207158 207252 207319 207403 207512 208464 208617 209984 210108 211085 211247 211749 212081 212108 212233 212400 212727 212753 212841 213928 214274 214292 214784 214905 215142 215155 215243 216325 216352 216593 216781 217490 217510 217602 217742 217789 218892 219437 219541 220409 220420 220635 220772 221006 221140 221190 221616 222004 222500 222600 223001 223309 223810 224702 225925 226084 226697 226704 227759 227850 229256 229431 229603 229671 229922 230378 230423 230481 230504 230636 230767 231124 231392 231402 231680 231775 232155 232787 233031 233054 233581 233689 233791 234751 234880 235388 236290 236404 236446 237014 237290 237694 237774 238059 238124 238267 238495 238547 238605 238631 238661 238675 239078 239856 240341 240588 241177 241203 241370 241877 242467 242603 242997 243303 243391 244468 245738 246329 247280 247919 248603 249013 249155 249464 250488 250530 250758 250763 250862 251234 251250 251752 251974 252368 252733 253040 253177 254702 254887 255412 255556 256231 256325 256335 256355 256359 256442 256455 256755 257424 257547 258261 258358 258413 259054 259583 260207 260256 260448 260455 260893 260908 260916 260940 261083 261275 261534 263199 263543 263621 264238 264305 264787 265171 265660 266014 266046 266698 266748 266853 266865 266891 267358 267840 267994 268090 268591 268867 270398 270406 271084 271571 271598 272498 272927 273101 273194 273230 273347 273504 274297 275024 275200 275441 275472 275717 275790 275896 276024 276102 276427 276726 276966 277276 277466 277493 277577 277812 278094 278647 279196 279405 279558 280094 280116 280168 280977 281411 281666 281858 282925 283085 283147 283532 283691 283864 284386 284488 284604 284876 285406 285520 286351 286412 286592 286808 286874 286955 287148 287389 287545 287775 288182 288667 289015 289102 290449 290620 290649 291545 291640 292036 292133 292647 292817 292930 293417 293778 293897 294300 294948 295273 295672 295700 295828 296718 296848 297002 297008 297179 297258 297298 297401 297705 297941 298033 298167 299042 299602 299737 300660 301036 301859 302034 302152 302647 302706 303119 303314 303323 303509 303877 304236 304472 306523 307636 307646 307783 307965 308142 308854 308910 309162 309311 309716 310116 310166 310476 310934 310941 310950 311366 311892 312405 312627 312887 313036 313330 313377 313831 313852 314839 315156 315207 315216 315263 316489 316758 317006 317307 317636 318251 318477 318792 319135 319279 319284 320164 320165 321351 321472 321553 321630 321663 321884 322901 323318 323798 324271 324324 325025 325598 325853 325881 326000 326147 326407 326512 326979 327112 327401 327753 328175 328796 329258 329512 329846 330389 330596 330912 330924 331366 331771 331862 332086 332232 333172 333237 333393 334177 334732 335578 335861 336420 336499 336559 337239 337251 337887 337968 338474 338508 338890 338998 339136 339312 339328 339730 340304 341243 341384 341412 341584 342315 342825 343126 343173 343308 343395 343621 343843 343900 343916 343951 344246 345483 346109 347044 347423 348224 348479 349503 349785 349830 349920 350297 350464 350536 351628 351712 352829 353136 353555 354517 354714 355085 356338 356541 356863 357401 358576 358731 358807 358920 359390 359695 359969 360256 361480 361569 361699 362226 362469 362481 362835 363332 364414 364425 365133 365931 366096 366202 366248 366500 366573 368033 368687 369070 369289 369356 369397 369686 370612 370664 370800 370858 371474 371564 371632 371850 372409 372501 372585 372721 373022 373152 373189 373910 373921 373994 374016 374196 374457 375114 375389 375436 375543 376043 376916 377211 377587 377808 377999 378249 378891 378988 379300 379429 379770 379799 379823 379865 380056 380249 380877 381132 381445 381824 381879 382349 382592 382875 383390 383747 383799 383918 384361 384947 385051 385111 386085 387114 387176 387321 388792 388809 389150 389306 389807 390169 390336 391251 391252 391963 392503 392579 392786 392848 392872 393160 393445 393798 393968 394383 394419 394476 394541 394799 394937 395021 395594 395627 395854 396852 397059 397570 397772 397861 398791 399312 399467 399896 400078 400141 400181 400201 400552 401034 401421 401497 401714 402474 403141 403329 404142 404853 404935 405072 405224 405381 406412 406412 406471 406940 406958 407273 407488 407741 407871 407975 408296 408303 408328 408419 408498 409069 409391 410014 410022 410643 410974 411228 411605 412689 412758 413604 414661 414677 414711 414821 415325 415453 415588 415813 415986 416107 416427 416578 416688 417251 417305 417730 417826 418353 419098 419369 419659 419670 419977 420224 421151 421785 423634 423911 424031 424261 424343 424392 424904 425069 425107 425299 425590 426185 426508 426528 426610 427114 427288 427797 427989 428324 429147 429336 429427 429511 429559 430018 430286 431252 431540 431817 432049 432145 432675 432775 433027 433067 433088 434419 434754 435605 435885 435955 436062 436611 436807 436875 437549 437752 438314 439898 440560 441232 441342 442439 443058 443060 443247 443256 443375 443603 443862 444579 444723 444735 444918 445143 445208 445409 446032 446454 446766 446804 447077 447396 447673 447774 447979 447993 448545 448890 449174 449297 449880 449998 450171 450485 450576 450769 451571 452293 452394 452707 453118 454289 454446 454582 455163 455184 456091 456357 456364 456405 456845 456894 457706 458130 458716 458994 459338 459727 459864 460014 460341 460631 461861 461916 462737 462766 463884 464978 465798 468029 468450 468767 468975 469182 469533 469755 470535 470645 471250 471463 471652 471718 471866 472006 472197 472691 473174 473256 473656 473902 474152 474474 474516 474596 474664 475104 475199 476990 478118 478302 478375 478779 480129 480486 481198 481414 482907 482916 483279 484029 484125 484217 484246 484254 484626 485055 485158 485408 485705 485796 486658 486853 486951 487018 487054 487112 487421 487795 488236 488417 488761 488946 489270 489356 489380 489570 489672 491029 491037 491079 491107 491815 492660 492983 493026 493190 493570 493966 494534 494847 495090 495568 495656 496462 496953 497781 498394 498806 498974 499260 499665 499754 499887 500361 500474 501102 501113 501143 501455 501479 502319 503191 503286 503313 503336 503392 503627 504113 504128 504506 504891 504969 505132 505224 505662 505710 505973 506693 506792 507120 507551 507632 507783 507962 508381 509668 509879 510288 510466 510791 510888 511110 512039 512206 512293 512510 512541 512588 512643 512645 513057 513271 513651 513818 515707 516771 516861 516978 517483 517759 517817 518113 519582 519636 520045 520277 520455 520474 520552 520631 520702 520704 521088 521162 521414 521437 522183 522462 522642 523748 523870 523881 524504 524591 526780 526869 527364 528414 529331 529492 529948 530586 530649 530692 530859 530956 531753 531939 532151 532167 532322 532754 533573 533785 533917 534923 535093 535126 535450 535612 536152 536209 536211 536702 536872 536873 537062 537800 538237 538638 538848 540067 540245 540418 540837 541280 541471 541566 542274 543084 544232 544906 545763 545869 546089 546249 546474 546561 546633 546709 547823 547933 548257 548741 548836 549683 550149 550804 553818 553854 555044 555099 555350 555388 555745 556425 556464 556723 556974 557458 557681 558209 558369 558831 558890 559181 559607 559851 559934 560032 561070 561151 561990 562639 562847 562892 563502 563647 563948 564406 564423 564456 564656 564707 565079 565123 565457 565908 566494 567315 567525 567730 567909 568041 568177 569178 569314 569391 569925 569981 570380 571235 571304 571425 571960 572494 572728 572851 572961 574374 574429 575419 576008 576676 576736 577449 577730 577960 578413 578725 578800 579110 579589 580352 580620 581001 581204 581421 582213 584209 584306 584391 585183 585244 585329 585366 586248 586263 586279 586321 586355 586643 586683 587460 588074 588435 589133 589403 589425 590384 590458 591727 592068 592120 592169 592477 592515 592833 592863 593680 594107 594282 594902 595363 595832 595913 596236 596267 596404 597034 597913 598502 599425 599458 599549 599553 600013 600399 600440 600508 600773 600816 600905 601047 601087 601148 601322 601359 601965 602031 602320 602842 603301 603863 604647 604798 604918 605407 605423 605741 605819 606347 606365 606398 606915 607228 607491 607518 607669 607843 607933 608418 608483 609087 609446 609536 609764 610356 610640 610653 610852 611041 611353 611840 612108 612169 612348 612627 612839 612881 612925 613550 614035 614159 614875 615440 615627 615993 616312 616427 616791 616957 617143 617327 618045 618329 619076 619246 619800 619803 621097 621280 621346 621380 621644 622354 622510 622640 622690 623253 623483 623682 623714 623971 624412 624714 624927 625472 626415 626477 626598 626705 627243 628220 628265 628601 628769 628926 629251 629305 630438 630624 630851 631536 631557 631608 632450 632546 632583 633109 633706 634029 634215 634388 634433 634731 634829 634910 635834 635944 636175 636326 636396 636527 636931 636992 637058 637954 638069 638247 638987 640410 640441 640573 640731 640910 641033 641605 642297 642866 642897 643090 643123 643146 643357 643896 643950 644247 644249 644517 644671 644786 645759 646023 646330 647651 647833 648182 648430 648999 649224 649551 649585 649622 649893 650028 650705 650797 651666 651864 651888 652118 652492 652580 652644 652734 653847 653918 653925 654554 654834 655265 655801 655929 656019 656063 656319 656709 656742 658383 658514 659419 659434 659451 659612 659862 660168 660221 660706 660793 662293 662301 662444 662691 662774 662902 663155 663500 663586 663671 664823 665030 665241 665885 665933 666227 667434 667636 667905 668033 668485 668525 668824 668896 668917 669672 669941 670319 670382 671235 671362 671453 671934 671970 672453 672956 673024 673414 673423 673568 673831 674017 674139 674159 674417 675223 675545 675545 675762 676129 677142 677638 677742 678639 679171 679931 680456 680632 680688 681516 683015 683091 683163 683405 683916 683938 684434 684922 685248 685265 685513 686127 686533 687474 687765 687836 689662 690661 690869 690900 691176 691722 692207 692351 692426 692570 692984 693542 693548 693621 693785 693790 694071 694082 694104 694123 694395 694446 694547 694825 694979 695730 696024 696295 696324 697852 697890 697992 697996 698802 698809 698878 699375 699469 700030 700360 700692 701068 701579 701919 702752 703322 703879 704335 704481 704561 704858 705007 705007 705175 705809 705903 706042 706113 706778 707552 707635 708616 709489 709587 709697 709859 710126 711300 711409 711493 712545 712868 713017 713062 713327 713485 713961 713982 714208 714294 714669 715114 715438 715470 715600 715980 716065 716475 716883 717049 717386 717565 718311 718687 718862 718912 719045 719103 719683 720153 720506 720819 720927 721323 722013 723885 724762 724946 726109 727066 727293 727635 727876 728095 728903 729028 729170 729331 729512 729557 730237 730763 731233 732390 733745 733786 734038 734197 734499 734952 735077 735475 735646 736737 736929 737314 737927 738701 738734 738738 738938 739328 739637 739711 740107 740257 740261 740795 741065 741163 741550 741632 742166 742438 742581 742899 742944 743263 743619 744350 744508 744675 746161 746703 746755 746990 747240 747758 747882 748310 748335 748350 749113 749402 749899 749956 750178 750358 750959 751154 751174 751305 751353 751468 751632 751635 752174 752213 752362 753147 753802 754387 754541 754782 754836 755155 755752 755770 755791 755864 756455 756659 757672 757722 757833 759491 759609 759727 759863 760239 760544 760965 761239 761533 761824 762025 762201 762272 762816 762858 763052 764077 764548 764829 764859 764989 765151 765558 765590 765752 766049 767239 768236 768855 768913 769267 769866 770273 771072 772680 774073 774250 774650 774835 775083 776103 776171 776199 776318 776640 777509 777688 777732 778060 779108 779584 779731 779959 780369 781373 782323 782527 782657 782988 783343 783352 783526 783882 784117 784596 785140 785203 785360 785381 785433 785497 785587 785772 786098 786125 786405 786541 786566 787153 787156 787425 788098 788925 789490 790073 790220 790466 790620 790638 790935 791271 791286 791528 791770 791818 792306 792339 792408 792427 792534 793319 793781 794455 795001 795165 795412 795428 795605 796706 797018 797482 797587 797981 798642 798669 798989 799021 799680 799928 799954 800151 800400 800554 800740 800836 800971 801046 801217 801483 801517 801901 802986 803067 803191 803354 803399 803465 803734 803965 804384 804538 804749 804902 804940 805456 805858 806336 806627 807055 807319 807772 807784 808630 808821 808903 809073 809394 809798 810048 810540 810674 811217 811414 811700 811837 812033 812102 812471 812962 813559 813671 813942 813957 814238 814347 814623 814687 814731 815901 816770 817915 817944 818818 819096 819548 819697 820218 820237 820820 820955 821182 821402 821415 821715 821743 822939 823093 823337 823660 824087 824436 824839 825403 825598 825773 826265 826658 826959 827446 828110 828681 829075 829164 829416 829852 829894 830382 831698 832304 832349 832554 832697 832817 832838 833784 833827 834284 834660 835083 835233 835715 836006 836012 836932 837245 837364 837664 838277 838492 838753 839082 839115 839461 840005 840218 840789 840934 840988 841453 841502 841575 841727 841843 841951 842043 842064 842081 842189 842303 843032 843825 844333 844727 844894 845184 845244 845251 845359 845695 845822 845868 846300 846474 846850 847494 848781 848841 849403 849637 849777 850392 850499 850577 851133 851891 851918 852121 852229 852479 852672 852727 853124 853409 853706 853712 854053 854414 854894 854958 855068 855106 856239 856465 857310 857379 857541 857618 857886 858030 858267 858794 859017 859338 859422 860117 860501 861010 862090 862168 862266 862506 862537 863427 863594 863742 864220 864223 864448 864864 865400 865402 865442 866124 866474 866538 866770 867722 868196 868383 869028 869048 870061 870406 870412 870649 870909 870989 872551 872707 873048 874976 875292 875304 875331 875746 875769 876213 877418 877567 878206 878594 878599 879053 879170 879449 879524 880132 880357 881232 881475 881520 882157 882159 882493 882723 883165 883752 883890 884560 884574 884621 884803 885194 885691 886146 886204 887297 888843 889708 890120 890121 890256 890857 891434 892151 892436 892581 893044 893211 893396 893580 893582 893592 893707 893841 893902 893945 894580 894931 895261 895277 895426 895458 896217 896991 898751 899160 899363 899655 899722 899986 900423 900630 901525 902491 902754 903203 903232 903325 904210 904556 904801 904952 905367 905681 906124 906620 906664 906774 906926 907539 907556 907852 908510 909359 909491 909563 909723 909740 910583 910665 910975 911130 911276 912215 912348 912761 912873 913131 913161 913499 913639 913678 914100 914103 914112 914568 914630 915667 915884 916355 916928 917079 917316 917544 917847 918318 918700 919325 919442 919654 920167 920273 921764 922403 922509 922542 922777 923418 923798 923983 924660 924967 925580 925808 926377 926443 926516 926538 926864 927225 927965 928094 928242 928635 930272 930417 930681 931215 931431 931860 932235 932272 932709 933917 934316 934877 935243 935743 935904 936191 936690 936774 938050 938235 939034 939238 939250 939458 940013 940674 941132 941348 941972 942177 942181 942193 942398 942576 942771 942969 943594 943830 943831 943984 944554 946092 946759 947312 947669 948171 949180 950323 950552 950988 951009 951253 951391 951729 952027 952408 952451 952560 953177 953435 953625 953689 954167 954419 954427 954659 955875 956465 956587 957704 957751 957755 957768 957881 958317 959547 959552 959698 960044 960233 960448 960622 960722 961164 961467 961572 961658 962539 963051 963128 963206 963418 963456 963490 963809 964052 964223 964350 964402 964519 965148 965411 965884 965978 966090 966696 967821 968081 968091 968098 968467 968705 968851 969090 969303 969488 969534 969751 971113 971858 972665 973193 974030 974283 975387 975396 975452 975632 975881 976119 976540 977339 977744 977804 978105 978253 978776 978784 979005 979867 980202 980298 980472 980637 980700 981403 981671 981731 982116 982333 982642 983554 983782 985027 985319 985326 986094 986948 987287 987354 987585 987952 988083 988428 988587 988630 988860 989192 991742 991930 992281 992753 993208 993270 993385 993420 993492 993569 993665 993957 994017 994110 994555 995002 995882 996246 996252 997590 997933 998077 998689 998883 999002 999183 999560 999843 999844 
600226 510810 92503 826150 952969 583138 547515 268176 684424 663419 184530 50412 173706 128123 129268 231528 385546 405163 537953 898055 412551 205857 141699 737575 783910 446975 909990 864282 159991 585716 780711 276568 96525 389565 102718 49494 972702 166584 317669 173478 830002 18550 223889 520060 146672 869508 751587 48569 274670 289539 462976 203573 11747 604674 941147 795657 51649 367488 176290 727991 953204 957000 4559 49728 346564 623628 
