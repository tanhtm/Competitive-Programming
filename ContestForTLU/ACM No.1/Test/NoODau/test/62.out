1768
1531
262
2457
2815
1720
1631
813
2037
1978
550
154
510
385
388
701
1167
1225
1609
2667
1250
624
420
2183
2315
1351
2701
2579
472
1727
2306
838
275
1175
294
149
2878
493
968
510
2468
55
680
1556
431
2593
2231
146
828
883
1397
618
32
1786
2784
2360
157
1102
520
2160
2816
2826
14
150
1054
1852
